security: extend overview to cover secure boot (#317)
foundriesio/docs: source/reference-manual/security/security.rst

.. _targets.json:
   https://theupdateframework.com/metadata/

Secure Boot (Hardware Root of Trust)
------------------------------------

> :ref:`Secure Boot <ref-secure-boot>` is the mechanism used to force a device to only execute boot software

Reviewer: maybe keep this under 80 columns (we might need to fix this in several parts of the docs)

Author: Yeah, was curious if the rendering would have any effect here but it doesn't break the line, let me fix it (and we should probably have a linter for this (@kprosise).

that is signed by a certain set of keys. The verification process and respective

> security functions is performed by the SoC boot ROM, and it is the starting point

Reviewer: ```suggestion
security functions are performed by the SoC boot ROM, and these are the starting points
```
